client/src/components/Dataset/DatasetView.vue

```diff
@@ -0,0 +1,384 @@
+<script setup lang="ts">
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faChevronDown, faChevronUp } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { BLink, BTab, BTabs } from "bootstrap-vue";
+import { computed, ref, watch } from "vue";
+import { useRouter } from "vue-router/composables";
+
+import { STATES } from "@/components/History/Content/model/states";
+import { usePersistentToggle } from "@/composables/persistentToggle";
+import { useDatasetStore } from "@/stores/datasetStore";
+
+import Heading from "../Common/Heading.vue";
+import DatasetError from "../DatasetInformation/DatasetError.vue";
+import LoadingSpan from "../LoadingSpan.vue";
+import DatasetAttributes from "@/components/DatasetInformation/DatasetAttributes.vue";
+import DatasetDetails from "@/components/DatasetInformation/DatasetDetails.vue";
+import VisualizationsList from "@/components/Visualizations/Index.vue";
+
+library.add(faChevronUp, faChevronDown);
+
+const datasetStore = useDatasetStore();
+const router = useRouter();
+const iframeLoading = ref(true);
+
+// Use persistent toggle for header state
+const { toggled: headerCollapsed, toggle: toggleHeaderCollapse } = usePersistentToggle("dataset-header-collapsed");
+const headerState = computed(() => (headerCollapsed.value ? "closed" : "open"));
+
+const props = defineProps({
+    datasetId: {
+        type: String,
+        required: true,
+    },
+    tab: {
+        type: String,
+        default: "preview",
+    },
+});
+
+const TABS = {
+    PREVIEW: "preview",
+    VISUALIZE: "visualize",
+    DETAILS: "details",
+    EDIT: "edit",
+    ERROR: "error",
+} as const;
+
+type TabName = (typeof TABS)[keyof typeof TABS];
+const TAB_VALUES = Object.values(TABS) as TabName[];
+
+const activeTab = ref<TabName>(TABS.PREVIEW);
+
+const dataset = computed(() => datasetStore.getDataset(props.datasetId));
+const isLoading = computed(() => datasetStore.isLoadingDataset(props.datasetId));
+const displayUrl = computed(() => `/datasets/${props.datasetId}/display/?preview=true`);
```

Comment:
This is different in #20154, but there we do need 
```suggestion
const displayUrl = computed(() => withPrefix(`/datasets/${props.datasetId}/display/?preview=true`));
```